crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -56,84 +140,233 @@ impl TryFrom<&JpmorganRouterData<&PaymentsAuthorizeRouterData>> for JpmorganPaym
-#[serde(rename_all = "lowercase")]
-pub enum JpmorganPaymentStatus {
-    Succeeded,
-    Failed,
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum JpmorganTransactionStatus {
+    Success,
+    Denied,
+    Error,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum JpmorganTransactionState {
+    Closed,
+    Authorized,
+    Voided,
     #[default]
-    Processing,
+    Pending,
+    Declined,
+    Error,
 }
 
-impl From<JpmorganPaymentStatus> for common_enums::AttemptStatus {
-    fn from(item: JpmorganPaymentStatus) -> Self {
+#[derive(Default, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganPaymentsResponse {
+    transaction_id: String,
+    request_id: String,
+    transaction_state: JpmorganTransactionState,
+    response_status: String,
+    response_code: String,
+    response_message: String,
+    payment_method_type: PaymentMethodType,
+    capture_method: Option<CapMethod>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Merchant {
+    merchant_id: Option<String>,
+    merchant_software: MerchantSoftware,
+    merchant_category_code: Option<String>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MerchantSoftware {
+    company_name: Secret<String>,
+    product_name: Secret<String>,
+    version: Option<Secret<String>>,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PaymentMethodType {
+    card: Option<Card>,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Card {
+    expiry: Option<ExpiryResponse>,
+    card_type: Option<Secret<String>>,
+    card_type_name: Option<Secret<String>>,
+    masked_account_number: Option<Secret<String>>,
+    card_type_indicators: Option<CardTypeIndicators>,
+    network_response: Option<NetworkResponse>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NetworkResponse {
+    address_verification_result: Option<Secret<String>>,
+    address_verification_result_code: Option<Secret<String>>,
+    card_verification_result_code: Option<Secret<String>>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExpiryResponse {
+    month: Option<Secret<i32>>,
+    year: Option<Secret<i32>>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CardTypeIndicators {
+    issuance_country_code: Option<Secret<String>>,
+    is_durbin_regulated: Option<bool>,
+    card_product_types: Secret<Vec<String>>,
+}
+
+impl From<JpmorganTransactionStatus> for common_enums::AttemptStatus {
+    fn from(item: JpmorganTransactionStatus) -> Self {
```

Comment:
Why is this function in code if it is not being used anywhere?

Reply:
removed